wasm/jit/jit_arm64.go

```diff
@@ -554,8 +622,228 @@ func (c *arm64Compiler) compileBrTable(o *wazeroir.OperationBrTable) error {
+
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		reservedRegisterForEngine, engineGlobalContextCallFrameStackPointerOffset,
+		tmpRegisters[1])
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD, reservedRegisterForEngine,
+		engineGlobalContextCallFrameStackElement0AddressOffset, tmpRegisters[2])
+	// Calculate "callFrameStackTopAddressRegister = tmpRegisters[2] + tmpRegisters[1] << $callFrameDataSizeMostSignificantSetBit".
+	c.emitAddInstructionWithLeftShiftedRegister(
+		tmpRegisters[1], callFrameDataSizeMostSignificantSetBit,
+		tmpRegisters[2],
+		callFrameStackTopAddressRegister,
+	)
+
+	// At this point, we have:
+	//
+	//    [..., ra.current, rb.current, rc.current, _, ra.next, rb.next, rc.next, ...]  <- call frame stack's data region (somewhere in the memory)
+	//                                              ^
+	//                              callFrameStackTopAddressRegister
+	//                (absolute address of &callFrame[engine.callFrameStackPointer]])
+	//
+	// where:
+	//      ra.* = callFrame.returnAddress
+	//      rb.* = callFrame.returnStackBasePointer
+	//      rc.* = callFrame.compiledFunction
+	//      _  = callFrame's padding (see comment on callFrame._ field.)
+	//
+	// In the following comment, we use the notations in the above example.
+	//
+	// What we have to do in the following is that
+	//   1) Set rb.current so that we can return back to this function properly.
+	//   2) Set engine.valueStackContext.stackBasePointer for the next function.
+	//   3) Set rc.next to specify which function is executed on the current call frame (needs to make Go function calls).
+	//   4) Set ra.current so that we can return back to this function properly.
+
+	// 1) Set rb.current so that we can return back to this function properly.
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		reservedRegisterForEngine, engineValueStackContextStackBasePointerOffset,
+		tmpRegisters[1])
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		tmpRegisters[1],
+		// "rb.current" is BELOW the top address. See the above example for detail.
+		callFrameStackTopAddressRegister, -(callFrameDataSize - callFrameReturnStackBasePointerOffset))
+
+	// 2) Set engine.valueStackContext.stackBasePointer for the next function.
+	//
+	// At this point, tmpRegisters[1] holds the old stack base pointer. We could get the new frame's
+	// stack base pointer by "old stack base pointer + old stack pointer - # of function params"
+	// See the comments in engine.pushCallFrame which does exactly the same calculation in Go.
+	c.applyConstToRegisterInstruction(arm64.AADD,
+		int64(c.locationStack.sp)-int64(len(functype.Params)),
+		tmpRegisters[1])
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		tmpRegisters[1],
+		reservedRegisterForEngine, engineValueStackContextStackBasePointerOffset)
+
+	// Alias for readability.
+	compiledFunctionAddressRegister := tmpRegisters[1]
```

Comment:
it is this sort of thing I meant, but ack [1] could be reused for many things, then only now forward used for this purpose. However, if they are always ever only used for one thing, best to alias early. If always temp well, it is what it is, but at least you'd know explicitly!

Reply:
pps I get that returning a slice is handy on cleanup vs tracking all of them individually. It may be nice to think if a defer function could ever work here. then you don't need to wander around looking if they are deallocated at the end.